cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -227,6 +234,9 @@ public ApplicationWithPrograms updateApp(Id.Application appId, AppRequest appReq
     if (currentSpec == null) {
       throw new ApplicationNotFoundException(appId);
     }
+    // App exists. Check if the current user has admin privileges on it before updating. The user's write privileges on
+    // the namespace will get enforced in the deployApp method.
+    authorizerInstantiatorService.get().enforce(appId.toEntityId(), SecurityRequestContext.toPrincipal(), Action.ADMIN);
```

Comment:
Didn't understand this comment : "The user's write privileges on the namespace will get enforced in the deployApp method."